airflow/jobs.py

```diff
@@ -1821,7 +1821,8 @@ def _execute(self):
 
             for ti in run.get_task_instances():
                 # all tasks part of the backfill are scheduled to run
-                ti.set_state(State.SCHEDULED, session=session)
+                if ti.state == State.NONE:
```

Comment:
@artwr good question. Should they be retried or should the user clear them first? 1.7.1 seems to re-evaluate them, but is a bit ambiguous in other areas: for example a task instance set to SKIPPED is being skipped again, while the underlying condition might have a different outcome now.

btw: failed tasks are being re-evaluated in my code as well, but will be immediately kicked out as they failed earlier.